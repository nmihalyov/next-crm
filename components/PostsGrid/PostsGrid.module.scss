.grid {
  display: flex;
  flex-wrap: wrap;
  margin-left: -10px;
  margin-right: -10px;
}

.empty {
  opacity: 0.5;
  width: 100%;
  text-align: center;
}