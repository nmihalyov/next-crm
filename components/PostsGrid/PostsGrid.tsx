import { FC } from 'react';
import { Empty } from 'antd';
import type { Post } from '../../types/post';
import styles from './PostsGrid.module.scss';

import PostCard from '../_ui/PostCard/PostCard';

const PostsGrid: FC<{
  posts: Post[]
}> = ({ posts }) => {
  return (
    <div className={styles.grid}>
      {posts.length
        ? posts.map(post => <PostCard key={post.id} {...post} />)
        : <Empty
          image={Empty.PRESENTED_IMAGE_SIMPLE}
          description="No posts yet"
          className={styles.empty} />
      }
    </div>
  );
};

export default PostsGrid;