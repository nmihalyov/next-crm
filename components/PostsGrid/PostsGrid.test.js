import PostsGrid from './PostsGrid';

describe('PostsGrid:', () => {
  const setUp = props => shallow(<PostsGrid {...props} />);
  const posts = [{userId: 1, id: 1, title: 'Title', body: 'Post body text'}];

  it('renders component if any posts', () => {
    const component = setUp({posts});

    expect(component).toMatchSnapshot();
  });

  it('renders empty element if no posts', () => {
    const component = setUp();
    const empty = component.find('Empty');

    expect(empty).toMatchSnapshot();
  });
});