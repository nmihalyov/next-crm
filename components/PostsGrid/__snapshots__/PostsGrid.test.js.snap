// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsGrid: renders component if any posts 1`] = `
<div
  className="grid"
>
  <Memo(PostCard)
    body="Post body text"
    id={1}
    key="1"
    title="Title"
    userId={1}
  />
</div>
`;

exports[`PostsGrid: renders empty element if no posts 1`] = `
<Empty
  className="empty"
  description="No posts yet"
  image={<Simple />}
/>
`;
